packages/db/tests/lib/test-updateGranuleAndFiles.js

```diff
@@ -0,0 +1,228 @@
+        granule_cumulus_id: pgGranule.cumulus_id,
+        file_name: pgGranule.granule_id + '.txt',
+        updated_at: new Date().toISOString(),
+        bucket: t.context.collectionId + '--bucket',
+        key: t.context.collectionId + pgGranule.granule_id + '/key-txt.pem',
+        path: t.context.collectionId + '/' + pgGranule.granule_id,
+      }),
+      fakeFileRecordFactory({
+        granule_cumulus_id: pgGranule.cumulus_id,
+        file_name: pgGranule.granule_id + '.cmr',
+        updated_at: new Date().toISOString(),
+        bucket: t.context.collectionId + '--bucket',
+        key: t.context.collectionId + pgGranule.granule_id + '/key-cmr.pem',
+        path: t.context.collectionId + '/' + pgGranule.granule_id,
+      })
+    );
+  }
+
+  t.context.pgFiles = await t.context.filePgModel.insert(knex, t.context.files);
+  // update 1/2 of the granules to be moved to the new collection
+  t.context.movedGranules.push(await simulateGranuleUpdate(knex, t.context.granules.slice(0, 5),
+    t.context.collection, t.context.collectionId, t.context.collectionId2));
+
+  // the other half will be unmoved but translated to an apiGranule
+  t.context.movedGranules.push(await simulateGranuleUpdate(knex, t.context.granules.slice(5),
+    t.context.collection, t.context.collectionId, t.context.collectionId));
+
+  t.context.movedGranules = t.context.movedGranules.flat();
+});
+
+test.after.always(async (t) => {
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
+test.serial('updateGranuleAndFiles successfully updates a partial list of granules based on the collectionId change', async (t) => {
+  const {
+    granuleIds,
+    granulePgModel,
+    movedGranules,
+    collectionId2,
+    collectionId,
+    collection,
+    collection2,
+    knex,
+  } = t.context;
+  await updateGranuleAndFiles(knex, movedGranules);
+
+  const returnedGranules = await Promise.all(granuleIds.map((id) =>
+    getUniqueGranuleByGranuleId(knex, id, granulePgModel)));
+
+  for (const granule of returnedGranules) {
+    const testCollection = granule.cumulus_id >= 5 ? collection : collection2;
+    const testCollectionId = granule.cumulus_id >= 5 ? collectionId : collectionId2;
+    const apiGranule = await translatePostgresGranuleResultToApiGranule(knex, {
+      ...granule,
+      collectionName: testCollection.name,
+      collectionVersion: testCollection.version,
+    });
+    // the movedGranules param only has 1/2 of the granules to be moved to collection 2
+    // here we can check based on the granule's cumulus id which collection it should be a part of
+    t.true(apiGranule.collectionId === testCollectionId);
+    for (const file of apiGranule.files) {
+      t.true(file.key.includes(testCollectionId));
+      t.true(file.bucket.includes(testCollectionId));
+    }
+  }
+});
+
+test.serial('updateGranuleAndFiles successfully updates a complete list of granules, 1/2 of which have already been moved', async (t) => {
```

Comment:
I think we need a more complete set of units testing all granule updates, given the method isn't written to update files and the collection ID, but *all* granule fields. 

If that 's not intentional and this method is intending to be limited to updating file locations and the collectionID, we should probably make the method more defensive somehow.

Reply:
@Nnaga1  to some degree that depends on what the intent is.   *As written* this I don't believe this method  enforces the ticket context (only update collection and file paths)  

  If the intent is to provide a method that takes an array of granule objects and updates it *only* utilizing translation conventions and ignoring API write business logic, then we should test that broader case. 
  
  If the intent is this helper is designed to just update those specific fields, we should update it to do only that and test.    
  
 This is important as this method is an exposed package method and creating a user contract.